packages/zapp/console/src/components/Literals/helpers.ts

```diff
@@ -0,0 +1,325 @@
+import { formatDateUTC, protobufDurationToHMS } from 'common/formatters';
+import { sortedObjectEntries, timestampToDate } from 'common/utils';
+import { Core } from 'flyteidl';
+import * as Long from 'long';
+import {
+  BlobDimensionality,
+  ProtobufListValue,
+  ProtobufStruct,
+  ProtobufValue,
+  SchemaColumnType,
+} from 'models/Common/types';
+
+// PRIMITIVE
+function processPrimitive(primitive: Core.IPrimitive) {
+  const type = (primitive as Core.Primitive).value!;
+  switch (type) {
+    case 'datetime':
+      return formatDateUTC(timestampToDate(primitive.datetime!));
+    case 'duration':
+      return protobufDurationToHMS(primitive.duration!);
+    case 'integer': {
+      return Long.fromValue(primitive[type] as Long).toNumber();
+    }
+    case 'boolean':
+    case 'floatValue':
+      return primitive[type];
+    case 'stringValue':
+    default:
+      return `${primitive[type]}`;
```

Comment:
if type s not one of the above will `primitive[type]` be a valid experession?

Reply:
oh yeah, i typed them all out for proper type checking but left this one out accidentally. these are all known types, anything that is not a valid one will default and return unknown